src/transformers/models/deberta/modeling_tf_deberta.py

```diff
@@ -0,0 +1,1520 @@
+# coding=utf-8
+# Copyright 2021 kamalkraj and The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" TF 2.0 BrandNewBERT model. """
+
+
+import math
+from typing import Dict, Optional, Sequence, Tuple, Union
+
+import numpy as np
+import tensorflow as tf
+import tensorflow.experimental.numpy as tnp
+
+import tensorflow_probability as tfp
+
+from ...activations_tf import get_tf_activation
+from ...file_utils import add_code_sample_docstrings, add_start_docstrings, add_start_docstrings_to_model_forward
+from ...modeling_tf_outputs import (
+    TFBaseModelOutput,
+    TFMaskedLMOutput,
+    TFQuestionAnsweringModelOutput,
+    TFSequenceClassifierOutput,
+    TFTokenClassifierOutput,
+)
+from ...modeling_tf_utils import (
+    TFMaskedLanguageModelingLoss,
+    TFModelInputType,
+    TFPreTrainedModel,
+    TFQuestionAnsweringLoss,
+    TFSequenceClassificationLoss,
+    TFTokenClassificationLoss,
+    get_initializer,
+    input_processing,
+    shape_list,
+)
+from ...utils import logging
+from .configuration_deberta import DebertaConfig
+
+
+logger = logging.get_logger(__name__)
+
+tnp.experimental_enable_numpy_behavior()
```

Comment:
This is possibly going to break at each new release of TF, knowing how they usually deal with "experimental" stuff.

Reply:
This is for using `take_along_axis`, TensorFlow variant of NumPy's take_along_axis. Which is the equivalent of the `torch.gather`

https://www.tensorflow.org/api_docs/python/tf/experimental/numpy/take_along_axis

Reply:
Shouldn't it be `tf`? Also, using another operation more supporter by TensorFlow to do the gather is probably safer. Especially since this looks like an API recently introduced but the minimal version of TF is 2.3.

Reply:
code will work with a minimum version of TF 2.4. I verified. But need to disable line 53 
 
 `torch.gather` like function is not available in TF STABLE API. I couldn't find anything on google also. Please share if you have anything similar implemented inside hugging face/transformers.

Reply:
I believe it should be possible to make a pure-Tensorflow version of `torch.gather` or `np.take_along_axis`, and I agree with the other maintainers that it's preferable to do that than to depend on `tf.experimental.numpy`, which will certainly get shaken up. Making awkward TF implementations of Torch gather layers is my nerd [special topic](https://github.com/tensorflow/addons/pull/2352), so let me try this and see what I can figure out.

Reply:
Hi, I have what I believe is a working implementation! Can you try this in your code in place of the existing `torch_gather` and let me know how it performs?
```
def tf_take_along_axis(x, indices, gather_axis):
    if gather_axis < 0:
        gather_axis = tf.rank(x) + gather_axis
    
    if gather_axis != tf.rank(x) - 1:
        pre_roll = tf.rank(x) - 1 - gather_axis
        permutation = tf.roll(tf.range(tf.rank(x)), pre_roll, axis=0)
        x = tf.transpose(x, perm=permutation)
        indices = tf.transpose(indices, perm=permutation)
    else:
        pre_roll = 0
        
    flat_x = tf.reshape(x, (-1, tf.shape(x)[-1]))
    flat_indices = tf.reshape(indices, (-1, tf.shape(indices)[-1]))
    gathered = tf.gather(flat_x, flat_indices, batch_dims=1)
    gathered = tf.reshape(gathered, tf.shape(indices))
    
    if pre_roll != 0:
        permutation = tf.roll(tf.range(tf.rank(x)), -pre_roll, axis=0)
        gathered = tf.transpose(gathered, perm=permutation)
        
    return gathered
```

Reply:
@Rocketknight1 The above implementation worked. Thanks